diff --git a/configure b/configure
index cdd4823..9e822b5 100755
--- a/configure
+++ b/configure
@@ -19275,16 +19275,12 @@ printf "%s\n" "$MACHDEP_OBJS" >&6; }
 fi
 
 if test "$ac_sys_system" = "Linux-android"; then
-  # When these functions are used in an unprivileged process, they crash rather
-  # than returning an error.
-  blocked_funcs="chroot initgroups setegid seteuid setgid sethostname
-    setregid setresgid setresuid setreuid setuid"
 
   # Before API level 23, when fchmodat is called with the unimplemented flag
   # AT_SYMLINK_NOFOLLOW, instead of returning ENOTSUP as it should, it actually
   # follows the symlink.
   if test "$ANDROID_API_LEVEL" -lt 23; then
-    blocked_funcs="$blocked_funcs fchmodat"
+    blocked_funcs="fchmodat"
   fi
 
   for name in $blocked_funcs; do
diff --git a/configure.ac b/configure.ac
index a5c670c..2401b44 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5239,16 +5239,12 @@ else
 fi
 
 if test "$ac_sys_system" = "Linux-android"; then
-  # When these functions are used in an unprivileged process, they crash rather
-  # than returning an error.
-  blocked_funcs="chroot initgroups setegid seteuid setgid sethostname
-    setregid setresgid setresuid setreuid setuid"
 
   # Before API level 23, when fchmodat is called with the unimplemented flag
   # AT_SYMLINK_NOFOLLOW, instead of returning ENOTSUP as it should, it actually
   # follows the symlink.
   if test "$ANDROID_API_LEVEL" -lt 23; then
-    blocked_funcs="$blocked_funcs fchmodat"
+    blocked_funcs="fchmodat"
   fi
 
   for name in $blocked_funcs; do
