diff --git a/refine/refine.in b/refine/refine.in
index c861f0c..64311b6 100755
--- a/refine/refine.in
+++ b/refine/refine.in
@@ -1,5 +1,4 @@
 #!@PYTHON@
-
 # refine.in
 #
 # Copyright 2024 Hari Rana (TheEvilSkeleton)
@@ -22,8 +21,6 @@
 import os
 import sys
 import signal
-import locale
-import gettext
 import tempfile
 import subprocess
 
@@ -31,13 +28,9 @@ from contextlib import suppress
 
 VERSION = '@VERSION@'
 pkgdatadir = '@pkgdatadir@'
-localedir = '@localedir@'
 
 sys.path.insert(1, pkgdatadir)
 signal.signal(signal.SIGINT, signal.SIG_DFL)
-locale.bindtextdomain('refine', localedir)
-locale.textdomain('refine')
-gettext.install('refine', localedir)
 
 if __name__ == '__main__':
     import gi
