--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -178,11 +178,9 @@ find_package(LibNotificationManager ${PROJECT_DEP_VERSION} CONFIG REQUIRED)
 find_package(LibTaskManager ${PROJECT_DEP_VERSION} CONFIG REQUIRED)
 find_package(LibColorCorrect ${PROJECT_DEP_VERSION} CONFIG REQUIRED)
-find_package(KWinDBusInterface CONFIG REQUIRED)
+find_package(KWinX11DBusInterface CONFIG REQUIRED)
-find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
 find_package(KRunnerAppDBusInterface CONFIG REQUIRED)
 find_package(KSMServerDBusInterface CONFIG REQUIRED)
 find_package(KF6ItemModels CONFIG REQUIRED)
-find_package(KSysGuard CONFIG REQUIRED)
 find_package(KF6KDED CONFIG REQUIRED)
 
 find_package(KF6Baloo ${KF6_MIN_VERSION})
