diff --git a/kcms/CMakeLists.txt b/kcms/CMakeLists.txt
index 43ad7d4..25cd690 100644
--- a/kcms/CMakeLists.txt
+++ b/kcms/CMakeLists.txt
@@ -1,6 +1,5 @@
 remove_definitions(-DQT_NO_CAST_FROM_ASCII -DQT_STRICT_ITERATORS -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_KEYWORDS)
 
-add_subdirectory(libkwindevices)
 
 if(X11_Xkb_FOUND AND XCB_XKB_FOUND)
     add_subdirectory( keyboard )
@@ -24,7 +24,6 @@ add_subdirectory(kded)
 add_subdirectory(runners)
 add_subdirectory(spellchecking)
 add_subdirectory(qtquicksettings)
-add_subdirectory(touchscreen)
 
 add_subdirectory(workspaceoptions)
 if (KF6Baloo_FOUND)
