From c35193fc54702d0a7637a99d8a33f41635b88817 Mon Sep 17 00:00:00 2001
From: Md Arif <111168803+sabamdarif@users.noreply.github.com>
Date: Sun, 20 Jul 2025 21:27:23 +0530
Subject: [PATCH] fix sqlite3_has_fts5 error

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index d6990eb..c4acec8 100644
--- a/meson.build
+++ b/meson.build
@@ -142,8 +142,8 @@ add_project_arguments(debug_cflags, language: 'c')
 libstemmer = cc.find_library('stemmer', required: get_option('stemmer'))
 have_libstemmer = libstemmer.found()
 
-if meson.is_cross_build() and not meson.has_exe_wrapper()
-  sqlite3_fts5 = meson.get_cross_property('sqlite3_has_fts5')
+if meson.is_cross_build() and not meson.can_run_host_binaries()
+  sqlite3_fts5 = meson.get_external_property('sqlite3_has_fts5', '')
   if sqlite3_fts5 == ''
     error('Please assign an appropriate value for sqlite3_has_fts5 in the [properties] section of your crossfile')
   elif sqlite3_fts5 == 'true'
-- 
2.48.1

