From 0d6ed4ae530ba7a09d80720d3a200212301b47f5 Mon Sep 17 00:00:00 2001
From: Md Arif <111168803+sabamdarif@users.noreply.github.com>
Date: Fri, 25 Jul 2025 13:12:44 +0530
Subject: [PATCH 16/18] remove polkit dependency for cc-hostname-entry

---
 panels/common/cc-hostname-entry.c | 40 ++++++++-----------------------
 1 file changed, 10 insertions(+), 30 deletions(-)

diff --git a/panels/common/cc-hostname-entry.c b/panels/common/cc-hostname-entry.c
index 29e3b98..1b392ce 100644
--- a/panels/common/cc-hostname-entry.c
+++ b/panels/common/cc-hostname-entry.c
@@ -83,41 +83,21 @@ cc_hostname_entry_dispose (GObject *object)
 static void
 cc_hostname_entry_constructed (GObject *object)
 {
-  CcHostnameEntry *self = CC_HOSTNAME_ENTRY (object);
-  GPermission *permission;
-  g_autoptr(GError) error = NULL;
-  g_autofree gchar *str = NULL;
-
-  permission = polkit_permission_new_sync ("org.freedesktop.hostname1.set-static-hostname",
-                                           NULL, NULL, NULL);
-
-  /* Is hostnamed installed? */
-  if (permission == NULL)
-    {
-      g_debug ("Will not show hostname, hostnamed not installed");
+    CcHostnameEntry *self = CC_HOSTNAME_ENTRY(object);
+    g_autofree gchar *str = NULL;
 
-      gtk_widget_set_sensitive (GTK_WIDGET (self), FALSE);
+      gtk_widget_set_sensitive(GTK_WIDGET(self), TRUE);
 
-      return;
+    str = cc_hostname_get_display_hostname(cc_hostname_get_default());
+    if (str != NULL) {
+        gtk_editable_set_text(GTK_EDITABLE(self), str);
+    } else {
+        gtk_editable_set_text(GTK_EDITABLE(self), "");
     }
 
-  if (g_permission_get_allowed (permission))
-    gtk_widget_set_sensitive (GTK_WIDGET (self), TRUE);
-  else
-    {
-      g_debug ("Not allowed to change the hostname");
-      gtk_widget_set_sensitive (GTK_WIDGET (self), FALSE);
-    }
-
-  str = cc_hostname_get_display_hostname (cc_hostname_get_default ());
-  if (str != NULL)
-    gtk_editable_set_text (GTK_EDITABLE (self), str);
-  else
-    gtk_editable_set_text (GTK_EDITABLE (self), "");
-
-  g_signal_connect (self, "apply", G_CALLBACK (text_changed_cb), NULL);
+    g_signal_connect(self, "apply", G_CALLBACK(text_changed_cb), NULL);
 
-  adw_entry_row_set_show_apply_button (ADW_ENTRY_ROW (self), TRUE);
+  adw_entry_row_set_show_apply_button(ADW_ENTRY_ROW(self), TRUE);
 }
 
 static void
-- 
2.48.1

